Fixed zero ideal issue

Sondzus opened the pull request:
When B={} it really should be the zero monomial in the appropriate ring.

Sondzus committed:
Fixed zero ideal issue

When B={} it really should be the zero monomial in the appropriate ring.

sdgpfo commented:
So, the package loads, and installs, but all the tests fail: going through to figure out why now.

Sondzus committed:
option for including zero ideals in base function of RMIs.m2

same as before:
idealsFromGeneratingSets(B,p,D,"testfile”)
new functionality:
idealsFromGeneratingSets(B,p,D,"testfile”,includeZeroIdeals=>true)

Sondzus commented:
which test? sorry i should have tested.

Sondzus commented:
@sdgpfo , i just ran the tests:
check "RandomMonomialIdeals" 
and nothing seems to have failed.

sdgpfo commented:
@Sondzus Maybe you forgot to installPackage first? (it's happened to me before)

sdgpfo commented:
Oh sorry, I posted, without seeing the rest of the conversation. Don't worry about this now, let's talk later (or I'll post again if I have something new to say)

sdgpfo commented:
Maybe you forgot to installPackage first?

But, I was wrong about apply-that line definitely works.
We did need the return statement.
The last test fails because the min degree of B could now be -infinity, not
1. I commended out that test, I don't think we need it, anymore.

On Wed, May 17, 2017 at 2:07 PM, SP <notifications@github.com> wrote:

> i just ran check "RandomMonomialIdeals" and nothing seems to have failed.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/RandCommAlg/RMI/pull/5#issuecomment-302197902>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ARvaOGE1QKpFd1MLF3tMMJMs14DsUYicks5r60VqgaJpZM4NeSfl>
> .
>

sdgpfo reviewed (changes requested):
We need to replace Line 95 by the following two lines, because we need the 0-monomial to be in a set also, and because we need to return B after apply is run.
    apply(#B,i-> if B_i==={} then B=replace(i,{0_R},B));
    return(B)
We will also have to update the Test checking for probability 0 events, as that check no longer works with this update.

Sondzus commented:
OK I will take a look again. I didn't forget to installPackage, not sure why my test didn't fail. 
Also one does not need return(B) if the last line before it without the semicolon actually prints out the B. So I'll check this and get back to you with an update! Thanks!!

Sondzus commented:
Without the return(B), this is the output I get from the method - it looks perfectly fine to me. So why do we need a return B line @sdgpfo ?


i156 : randomGeneratingSets(2,3,0.1,10)

                    2           3     3         2       2   3       2       2
o156 = {{x x }, {x x }, {x }, {x }, {x }, {}, {x }, {x x , x }, {x x }, {x x }}
          1 2     1 2     1     2     2         2     1 2   2     1 2     1 2

Sondzus committed:
requested edits done

- added return(B) since apply output was weird
- fixed the 0 probability check to make it work

Sondzus commented:
@sdgpfo  I fixed the things you suggested,thanks.

Sondzus committed:
deleted firstFunction and other superfluous comments

this was really draft garbage.

Sondzus committed:
0 turned into {0} in randomGeneratingSets

also fixed the corresponding failing test.

Sondzus committed:
small output edit in printing # zero ideals in non-package m2 file.

Sondzus committed:
optimized dealing with 0 ideals in idealsFromGeneratingSets

streamlined some lines and added an internal function for extracting
non-zero ideals to be used in other methods throughout this file.
